tilespec.ts

```diff
@@ -83,7 +242,14 @@ function parseGridSizeIgnoringErrors(s: string): GridSize|null {
 }
 
 export class XY {
-    constructor(readonly x: number, readonly y: number) {}
```

Comment:
This change makes XY mutable, which is undesirable. Do you use the mutability somewhere? If not, please revert. If so, I suggest finding another way.